src/main/java/uk/go/hm/icb/dto/ICBMatch.java

```diff
@@ -40,9 +26,13 @@ public ICBMatchBuilder matches(String firstNameMatched, String lastNameMatched,
                     Pair.of("Address", addressMatched),
                     Pair.of("Unique Identifier - Birth Cert", birthCertificateMatched),
                     Pair.of("Driving Licence Number", drivingLicenseNumberMatched),
-                    Pair.of("Passport Number", passportNumber),
-                    Pair.of("Flag", flag)
+                    Pair.of("Passport Number", passportNumber)
             );
+            if (StringUtils.hasText(flag)) {
```

Comment:
Previous code eliminates this null check as the caller always knows what they are calling and getting back.

Reply:
The problem is that for LEV records even if there's no flag, we still have to pass in `-` and is displayed at UI.

Reply:
What about this logic?

public ICBMatchBuilder matches(String firstNameMatched, String lastNameMatched, String middleNameMatched,
                                       String dateOfBirthMatched, String addressMatched, String birthCertificateMatched,
                                       String drivingLicenseNumberMatched, String passportNumber) {
            matches = List.of(
                    Pair.of("First Name", firstNameMatched),
                    Pair.of("Last Name", lastNameMatched),
                    Pair.of("Middle Name", middleNameMatched),
                    Pair.of("Date of Birth", dateOfBirthMatched),
                    Pair.of("Address", addressMatched),
                    Pair.of("Unique Identifier - Birth Cert", birthCertificateMatched),
                    Pair.of("Driving Licence Number", drivingLicenseNumberMatched),
                    Pair.of("Passport Number", passportNumber)
            );
            return this;
        }

        public ICBMatchBuilder matches(String firstNameMatched, String lastNameMatched, String middleNameMatched,
                                       String dateOfBirthMatched, String addressMatched, String birthCertificateMatched,
                                       String drivingLicenseNumberMatched, String passportNumber, String flag) {
            ICBMatchBuilder builder = matches(firstNameMatched, lastNameMatched, middleNameMatched, dateOfBirthMatched, addressMatched, birthCertificateMatched, drivingLicenseNumberMatched, passportNumber);
            builder.matches.add(Pair.of("Flag", flag));
            return builder;
        }

Reply:
List.of() is immutable - so you can't do add again - that's why I recreated List in my code.

Reply:
I missed that point. Then let's make the original list mutable

```
 public ICBMatchBuilder matches(String firstNameMatched, String lastNameMatched, String middleNameMatched,
                                       String dateOfBirthMatched, String addressMatched, String birthCertificateMatched,
                                       String drivingLicenseNumberMatched, String passportNumber) {
            matches = new ArrayList<>(List.of(
                    Pair.of("First Name", firstNameMatched),
                    Pair.of("Last Name", lastNameMatched),
                    Pair.of("Middle Name", middleNameMatched),
                    Pair.of("Date of Birth", dateOfBirthMatched),
                    Pair.of("Address", addressMatched),
                    Pair.of("Unique Identifier - Birth Cert", birthCertificateMatched),
                    Pair.of("Driving Licence Number", drivingLicenseNumberMatched),
                    Pair.of("Passport Number", passportNumber)
            ));
            return this;
        }
```

I am trying to replace if-else with polymorphism. that's all.